### PR TITLE
Implement and enable redirector eTag adapter.

### LINKED ISSUE #3026: Prepared downloads via redirector should flush cache
When a document gets checked out and the *.zem download is prepared via [Redirector](https://github.com/4teamwork/opengever.core/blob/master/opengever/base/redirector.py#L56), the cache has to be flushed, otherwise the file is not downloaded until the user opens a not cached site. 

Solution:
 - [ ] Include the redirector cookie in the cachekey, with a separate ETag Adapter.

### PR DESCRIPTION
**Pull request by phgross**
Flush cache when preparing a download via Redirector adapter.

Fixes #3026

### CODE REVIEW

File: opengever/base/redirector.py [hunk 1]
```diff
@@ -122,3 +131,22 @@ def render(self):
...
+
+    def compute_md5hash(self, value):
+        m = hashlib.md5()
+        m.update(value.encode('utf-8'))
+        return m.hexdigest()
+
+    def __call__(self):
+        value = RedirectorCookie(self.request)._get_value()
+        if value:
+            return self.compute_md5hash(value)
```
Comment (deiferni) [inline opengever/base/redirector.py]: i'd add explicit returns for all paths for readability, so either add `return None` at the end or implement guard-clause style (which i'd prefer):

```python
value = RedirectorCookie(self.request)._get_value()
if not value:
    return None
return self.compute_md5hash(value)
```

File: opengever/base/redirector.py [hunk 2]
```diff
@@ -26,7 +30,7 @@ def read(self, remove=True):
             self._set([])
         return value
 
-    def _get(self):
+    def _get_value(self):
```
Comment (deiferni) [inline opengever/base/redirector.py]: this is still a protected method, but now it is used publicly. i'd drop the leading `_`.

File: opengever/base/tests/test_redirector.py [hunk 3]
```diff
@@ -98,3 +102,30 @@ def test_viewlet_only_redirects_once(self):
...
+
+        self.assertIsNone(self.get_etag_value_for(document))
+
+        # register redirect
+        redirector = IRedirector(self.request)
+        redirector.redirect(
+            'http://nohost/plone/document-1/externaledit',
+            target='named-window')
+
+        self.assertEquals('bcda738beecd660241c93d98dc016a9c',
```
Comment (deiferni) [inline opengever/base/tests/test_redirector.py]: think it is a bit brittle to assert against the cookie hash here, may be good enough to assert that we actually get a non-None value?

**Comment by phgross** [comment]
@deiferni updated ...

**Pull request merged**